### DISCUSSION TITLE
Level 22 Dex - Sentry 429 (Too Many Requests) in MetaMask

### DISCUSSION THREAD

**Discussion by remedcu** [OpenZeppelin/ethernaut#565] [Support]
I have created an issue in MetaMask, which you can see here: https://github.com/MetaMask/metamask-extension/issues/15000

This happens in `approve` transactions, and I use the command: `await contract.approve(instance, 10000)`

I even imported the contract to remix, deployed it to the instance address, and tried the GUI for approval, but still the same result.

Out of curiosity, I tried another level (23 and so on), which works well.

Is it a problem with Metamask, OpenZeppelin, or some weird bug just for me?

**Comment by mattjaf** [comment]
There's other ways to interact with a contract. I noticed this as well and was still able to solve it.

I created a pull request on this issue adding a hint, I didn't want to be so obvious about it, I could of hinted at what does "at address do???" I might do that

**SPOILER ALERT**
Head over to Remix.
Create a new contract called `AnyToken.sol` with the contents of:
`import "https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol";`
compile it
Then, right under the deploy button there's a button that says `At Address`
paste the address of `await contract.token1()` and click
After that, paste the address of `await contract.token2()` and click

I'm sure you can figure out the rest.

I was stoked to find out that was possible
